>From dd7b60948caa34ca8d7af8bc4d8a5a4db68ca2ec Mon Sep 17 00:00:00 2001
From: Daniel Kahn Gillmor <dkg@fifthhorseman.net>
Date: Thu, 6 Nov 2014 12:08:31 -0500
Subject: [PATCH] explicitly dearmor, rather than --import upstream signing key

gnupg 2.1 handles --import by generating a pubring.kbx file, not a
pubring.gpg file.  So when gnupg 2.1 is installed verifying upstream
signatures failed when the upstream signing key was ascii-armored.

With this patch, we explicitly dearmor the .asc, rather than relying
on side effects that aren't true for all gpg versions.
---
 scripts/uscan.pl | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/scripts/uscan.pl b/scripts/uscan.pl
index 4456a7c..8c79b3a 100755
--- a/scripts/uscan.pl
+++ b/scripts/uscan.pl
@@ -872,9 +872,10 @@ sub process_watchline ($$$$$$)
 		}
 		# Need to convert an armored key to binary for use by gpgv
 		$gpghome = tempdir(CLEANUP => 1);
-		spawn(exec => [$havegpg, '--homedir', $gpghome, '--no-options', '-q', '--batch', '--no-default-keyring', '--import', $keyring],
+		my $newkeyring = "$gpghome/trustedkeys.gpg";
+		spawn(exec => [$havegpg, '--homedir', $gpghome, '--no-options', '-q', '--batch', '--no-default-keyring', , '--output', $newkeyring, '--dearmor', $keyring],
 		      wait_child => 1);
-		$keyring = "$gpghome/pubring.gpg";
+		$keyring = $newkeyring
 	    }
 	}
 
-- 
2.1.1

